tensorboard/pip_package/build_pip_package.sh

```diff
@@ -135,9 +115,66 @@ pip install -qU wheel 'setuptools>=36.2.0'
 python setup.py bdist_wheel --python-tag py2 >/dev/null
 python setup.py bdist_wheel --python-tag py3 >/dev/null
 
-if [ "$run_smoke_test" = 1 ]; then
-  smoke 2
-  smoke 3
-fi
+smoke() {
+  pymajorversion="$1"
+  if [ -z "${pymajorversion}" ]; then
+    pymajorversion="$(python -c 'import sys; print(sys.version_info[0])')"
+  fi
+  smokepython="python$1"
+  smokevenv="smoke-venv$1"
+  smoketf="$2"
+  set +x
+  printf '\n\n%70s\n' | tr ' ' '='
+  printf "Smoke testing with ${smokepython} and ${smoketf}...\n\n"
```

Comment:
These should be format specifiers rather than part of the format string:

```sh
printf 'Smoke testing with %s and %s...\n' "${smokepython}" "${smoketf}"
```

Reply:
Honest question: why is that better?  It's just harder to keep in sync both from a code standpoint and logically since you need to match the %s with the arguments, and it's more verbose.  When you have native string interpolation, why not use it?

Reply:
If you prefer

    printf '%s\n\n' "Smoke testing with ${smokepython} and ${smoketf}..."

then that’s also totally fine with me. But you don’t interpolate into
format strings (e.g., `smokepython` or `smoketf` could have percent
signs, which would screw everything up).

Reply:
Fair enough, I wasn't considering them containing format specifiers.  Changed to use echo.

Reply:
Okay. Just to share where I’m coming from: I personally always avoid
`echo` because it is so ill-defined. The `echo -n` behavior is
unportable, but you can’t use `--` in the way that you can with standard
utilities. The behavior of `echo -e '\x1b'` is unspecified (in fact, the
behavior is implementation-defined if _any_ argument contains a
backslash), but the behavior of `echo -e foo` _is_ specified and most
implementations get it wrong. Implicitly inserting single spaces is not
the end of the world, but I’d prefer to do that explicitly. Et cetera.

I am much happier sticking to `printf` for all my printing needs, but as
a rule I don’t usually request that others do the same.

Reply:
Oh, TIL that the POSIX spec actually recommends that new applications
use `printf` instead of `echo`:
<https://pubs.opengroup.org/onlinepubs/9699919799/utilities/echo.html#tag_20_37_16>

Reply:
Feel free to update it to use `printf '%s\n'` instead.  I think shell scripts are the wrong tool for the job if we're getting really concerned about how we handle pathological inputs, so I'm not inclined to worry about this much further myself.